### PR TITLE
Add report commands to execute commands after tests are finished

### PR DESCRIPTION
**Pull request by ANeumann82**
Signed-off-by: Andreas Neumann <aneumann@mesosphere.com>

**What this PR does / why we need it**:
- Add report commands which are executed after the test steps are done, but before the cluster is stopped.

Fixes #

### CODE REVIEW

File: pkg/test/utils/kubernetes.go [hunk 1]
```diff
@@ -1057,8 +1067,6 @@ func RunCommands(logger Logger, namespace string, commands []harness.Command, wo
 	}
 
 	for _, cmd := range commands {
-		logger.Logf("running command: %q", cmd.Command)
```
Comment (kensipe) [inline pkg/test/utils/kubernetes.go:1060]: why is this removed... this keeps coming and going... and was resolved with marcin a couple of weeks ago.  It likely we need to review those closed PRs.

**Comment by ANeumann82** [comment]
Yeah, all good points - This PR isn't supposed to be merged soon (or even at all), it's more an exploratory development. I was trying to integrate `kudo diagnostics collect` into KUDO Cassandra and needed a hook to run that after the tests were finished. 

I agree, we should focus on life-cycle support, I've also noticed that these `reportCommands` that are executed after all tests are ran are not necessarily what I need. A postTest command for each test case would be more what I need
